lib/kitchen/directions/bake_generic_eoc_section/v1.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Kitchen::Directions::BakeGenericEocSection
+  class V1
+    renderable
+
+    def bake(chapter:, metadata_source:, klass:, append_to: nil, uuid_prefix: '.',
+             include_intro: true, &block)
+      section_clipboard = Kitchen::Clipboard.new
+      sections = if include_intro
+                   chapter.pages.search("$.#{klass}")
+                 else
+                   chapter.non_introduction_pages.search("$.#{klass}")
+                 end
+      sections.each(&block) # transforms each section according to the given block
+      sections.cut(to: section_clipboard)
```

Comment:
`[enumerator].each` and `[enumerator].cut` is probably running the search twice.  It might be a little more efficient to cut the sections and then iterate on them inside the clipboard ([which is itself enumerable](https://github.com/openstax/kitchen/blob/main/lib/kitchen/clipboard.rb#L8)):

```ruby
pages = include_intro_page ? chapter.pages : chapter.non_introduction_pages
section_clipboard = pages.search(section_selector).cut
section_clipboard.each(&block)
```

A benefit of using these 3 lines to replace lines 9 through 16 is that you don't have to repeat the `search(section_selector)` call.

Reply:
There seems to be some kind of bug that's affecting this implementation: when I call `section.ancestor(:page).count_in(:chapter)` from the `block`, if the `section` has been cut to a clipboard, then the chapter count always returns the same number (i.e. if there are 3 pages in the chapter it always returns `3`, if there are 2 it always returns `2`, etc). Digging into this a little, I've discovered that `section.ancestor(:page)` does seem to return the correct page, so I've been stumped so far on why this page seems to always be getting counted as the last page of the chapter.

This affects the section content transform for Further Research, since those sections get a header link added that contains the `chapter.page` count of the section's origin page. I think a couple of other eoc sections (that are yet to be refactored) also do things with the section's origin page. Until we figure out this bug, I think the block has to be passed in before the sections get cut to the clipboard.

Reply:
ok, if don't fix now, you may make a PR with a failing spec so we can get to it later.